gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientReadChannel.java

```diff
@@ -0,0 +1,550 @@
+    this.readOptions = readOptions;
+    this.objectSize = itemInfo.getSize();
+    this.contentReadChannel = new ContentReadChannel(readOptions, resourceId);
+  }
+
+  @Override
+  public int read(ByteBuffer dst) throws IOException {
+    throwIfNotOpen();
+
+    // Don't try to read if the buffer has no space.
+    if (dst.remaining() == 0) {
+      return 0;
+    }
+    logger.atFiner().log(
+        "Reading %d bytes at %d position from '%s'", dst.remaining(), currentPosition, resourceId);
+    if (currentPosition == objectSize) {
+      return -1;
+    }
+    return contentReadChannel.readContent(dst);
+  }
+
+  @Override
+  public int write(ByteBuffer src) throws IOException {
+    throw new UnsupportedOperationException("Cannot mutate read-only channel: " + this);
+  }
+
+  @Override
+  public long position() throws IOException {
+    return currentPosition;
+  }
+
+  /**
+   * Sets this channel's position.
+   *
+   * <p>This method will throw an exception if {@code newPosition} is greater than object size,
+   * which contradicts {@link SeekableByteChannel#position(long) SeekableByteChannel} contract.
+   * TODO(user): decide if this needs to be fixed.
+   *
+   * @param newPosition the new position, counting the number of bytes from the beginning.
+   * @return this channel instance
+   * @throws FileNotFoundException if the underlying object does not exist.
+   * @throws IOException on IO error
+   */
+  @Override
+  public SeekableByteChannel position(long newPosition) throws IOException {
+    throwIfNotOpen();
+
+    if (newPosition == currentPosition) {
+      return this;
+    }
+
+    validatePosition(newPosition);
+    logger.atFiner().log(
+        "Seek from %s to %s position for '%s'", currentPosition, newPosition, resourceId);
+    currentPosition = newPosition;
+    return this;
+  }
+
+  @Override
+  public long size() throws IOException {
+    return objectSize;
+  }
+
+  @Override
+  public SeekableByteChannel truncate(long size) throws IOException {
+    return null;
+  }
+
+  @Override
+  public boolean isOpen() {
+    return channelIsOpen;
+  }
+
+  @Override
+  public void close() {
+    if (!channelIsOpen) {
+      logger.atFiner().log("Ignoring close: channel for '%s' is not open.", resourceId);
+      return;
+    }
+    logger.atFiner().log("Closing channel for '%s'", resourceId);
+    channelIsOpen = false;
+    contentReadChannel.closeContentChannel();
+  }
+
+  private class ContentReadChannel {
+
+    private static final int SKIP_BUFFER_SIZE = 8192;
+
+    // Size of buffer to allocate for skipping bytes in-place when performing in-place seeks.
+    private final int footerSize;
+    private final BlobId blobId;
+
+    // This is the actual current position in `contentChannel` from where read can happen.
+    // This remains unchanged of position(long) method call.
+    private long contentChannelCurrentPosition = -1;
+    private long contentChannelEnd = -1;
```

Comment:
nit: as these vars defined inside `ContentReadChannel` class, `contentChannel` prefix is redundant and we can remove it and rename these vars to `currentPosition` and `endPosition`.

Reply:
We already have `currentPosition` which tracks the current position from which read should start. IMO, although `contentChannel` is redundant It does help in readability.